### DISCUSSION TITLE
Allow editing (including comments) of xmake.conf

### DISCUSSION THREAD

**Discussion by matty0ung** [xmake-io/xmake#3114] [Ideas]
### Is your feature request related to a problem? Please describe.

Imagine that you want to use a different directory than `build` for the build.  Maybe you are trying out Xmake as a possible alternative build system for an existing project, and you don't want it making modifications in the existing `build` directory that the current build system uses.

After poking around, you discover that (on Linux) `.xmake/linux/x86_64/xmake.conf` contains a useful-looking setting `buildir = "build"`.  Maybe this is build directory.  So you add a comment `-- Changed to not clash with old build system` and edit the line to buildir = "xbuild".  Now you run xmake again and it removes the comment from the file.

### Describe the solution you'd like

I don't necessarily mind that the config files are in hidden directories, but it is painful that they are not editable.   Can we allow edits and comments in the file and have xmake respect them?  (Also, it would be good for the first line of the .conf file always to be a comment. Then you would not have to guess what the format is for comments!)

### Describe alternatives you've considered

I had a look at https://github.com/xmake-io/xmake/issues/820 but it didn't seem to be quite the same issue.

**Comment by SirLynix** [comment]
xmake.conf is an internal file used by xmake to retains its configuration across runs, it's not made to be edited (but you can export/import it using xmake config --export/import).

You can configure the buildir using `xmake f --builddir=xbuild`, this is [documented here](https://xmake.io/#/manual/builtin_variables?id=varbuildir).

**Comment by matty0ung** [comment]
OK, I see.  It's a shame that the documentation implies that `buildir` is a read-only variable.  But I see `xmake f --help` gives the info.

By the way, could it be possible to have `builddir` or `build_dir` as a synonym for `buildir`?  The latter seems wrong to me.

**Comment by waruqi** [comment]
It is not a read-only variable. just `xmake f --buildir=xxxx`

**Comment by qudix** [comment]
`buildir` is indeed a mispelling, not sure what can be done to fix that however (maybe deprecation?), not really that big a deal.

**Comment by waruqi** [comment]
I know it's the wrong spelling because I don't like `builddir`, which has two repeated `dd` characters, so I've simplified it to `buildir`. maybe it's just a matter of my personal habits, but I don't really want to change it just yet.

**Comment by SirLynix** [comment]
for english speakers (and I think it's even worse for native speakers) it's a bit misleading, maybe introducing `build_dir` as an alias of `buildir` would be a good idea here? same for other similar options

**Comment by matty0ung** [comment]
I wouldn't exactly say changing `builddir` to `buildir` is a simplification.  I mean it's one character less to type and beneficial if you don't like double Ds, but it arguably increases complexity because it's unexpected/misleading.

FWIW I there is an advantage to having a clear consistent naming convention, eg using [snake_case](https://en.wikipedia.org/wiki/Snake_case) or [camelCase](https://en.wikipedia.org/wiki/Camel_case).
